[23.0] Mandrel builds with 17.0.11 EA fail integration tests due to birth time change

## Description
We see JDK 17.0.11 ea-based failures in all integration tests due to [JDK-8316304](https://bugs.openjdk.org/browse/JDK-8316304) which is newly backported to `17.0.11+1` and better.

Failures look like this:

```
[INFO] Running org.test.NativeHelloResourceIT
2024-02-17 02:52:22,318 WARN  [io.qua.agr.dep.AgroalProcessor] (build-5) The Agroal dependency is present but no JDBC datasources have been defined.
2024-02-17 02:52:22,554 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-12) Hibernate ORM is disabled because no JPA entities were found
Executing "/home/runner/work/mandrel/mandrel/quarkus/integration-tests/jpa-without-entity/target/quarkus-integration-test-jpa-without-entity-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/home/runner/work/mandrel/mandrel/quarkus/integration-tests/jpa-without-entity/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.native.native-image-xmx=13g"
Feb 17, 2024 2:52:23 AM io.quarkus.runtime.ApplicationLifecycleManager run
ERROR: Failed to start application (with profile [prod])
java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.NoSuchFieldError: sun.nio.fs.UnixFileAttributes.st_birthtime_sec
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getFieldID(JNIFunctions.java:1279)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetFieldID(JNIFunctions.java:432)
	at java.base@17.0.11-beta/sun.nio.fs.UnixNativeDispatcher.init(Native Method)
	at java.base@17.0.11-beta/sun.nio.fs.UnixNativeDispatcher.<clinit>(UnixNativeDispatcher.java:668)
	at java.base@17.0.11-beta/sun.nio.fs.UnixFileSystemProvider.isDirectory(UnixFileSystemProvider.java:521)
	at java.base@17.0.11-beta/java.nio.file.Files.isDirectory(Files.java:2318)
	at io.smallrye.config.DotEnvConfigSourceProvider.getDotEnvFile(DotEnvConfigSourceProvider.java:63)
	at io.smallrye.config.DotEnvConfigSourceProvider.<init>(DotEnvConfigSourceProvider.java:22)
	at io.quarkus.runtime.configuration.RuntimeConfigBuilder.configBuilder(RuntimeConfigBuilder.java:24)
	at io.smallrye.config.SmallRyeConfigBuilder.lambda$build$0(SmallRyeConfigBuilder.java:722)
	at java.base@17.0.11-beta/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base@17.0.11-beta/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.base@17.0.11-beta/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
	at java.base@17.0.11-beta/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base@17.0.11-beta/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base@17.0.11-beta/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base@17.0.11-beta/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base@17.0.11-beta/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:722)
	at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
	at io.quarkus.runtime.generated.Config.createRunTimeConfig(Unknown Source)
	at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
	... 7 more

Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.205 s <<< FAILURE! -- in org.test.NativeHelloResourceIT
Error:  org.test.NativeHelloResourceIT.testHelloEndpoint -- Time elapsed: 0.012 s <<< ERROR!
```

Example failure in weekly CI:
https://github.com/graalvm/mandrel/actions/runs/7938323570/job/21677410793#step:12:3244

--- a/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationJavaNio.java
+++ b/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationJavaNio.java
@@ -24,6 +24,7 @@
  */
 package com.oracle.svm.hosted.jdk;
 
+import java.lang.reflect.Field;
 import java.net.InetAddress;
 import java.net.SocketAddress;
 import java.nio.ByteBuffer;
@@ -38,6 +39,7 @@ import org.graalvm.nativeimage.impl.InternalPlatform;
 import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
 import com.oracle.svm.core.feature.InternalFeature;
 import com.oracle.svm.core.jdk.JNIRegistrationUtil;
+import com.oracle.svm.util.ReflectionUtil;
 
 /**
  * Registration of classes, methods, and fields accessed via JNI by C code of the JDK.
@@ -155,8 +157,13 @@ public class JNIRegistrationJavaNio extends JNIRegistrationUtil implements Inter
         RuntimeJNIAccess.register(fields(a, "sun.nio.fs.UnixFileAttributes",
                         "st_mode", "st_ino", "st_dev", "st_rdev", "st_nlink", "st_uid", "st_gid", "st_size",
                         "st_atime_sec", "st_atime_nsec", "st_mtime_sec", "st_mtime_nsec", "st_ctime_sec", "st_ctime_nsec"));
-        if (isDarwin()) {
+        // See JDK-8316304, new in JDK 17.0.11+1
+        Field unixCreationTimeField = ReflectionUtil.lookupField(true, clazz(a, "sun.nio.fs.UnixFileAttributes"), "st_birthtime_nsec");
+        if (isDarwin() || unixCreationTimeField != null) {
             RuntimeJNIAccess.register(fields(a, "sun.nio.fs.UnixFileAttributes", "st_birthtime_sec"));
+        }
+        if (unixCreationTimeField != null) {
+            RuntimeJNIAccess.register(fields(a, "sun.nio.fs.UnixFileAttributes", "st_birthtime_nsec"));
         }
 
         RuntimeJNIAccess.register(clazz(a, "sun.nio.fs.UnixFileStoreAttributes"));

